azure-proto-core-test/IdentityKindTests.cs

```diff
@@ -0,0 +1,187 @@
+using azure_proto_core;
+using NUnit.Framework;
```

Comment:
Find out if there are exclusion lists of characters for valid identity kinds, add test cases to assert either way